opentelemetry-api/src/opentelemetry/resources/__init__.py

```diff
@@ -11,3 +11,45 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Dict, Union
+
+
+class Resource(ABC):
+    """This the interface that resources must implement"""
+    def __init__(self, labels: Dict[str, str]):
+        """Construct a resource.
+
+        Direct calling of the constructor is discouraged, as it cannot
+        take advantage of caching and restricts to the type of object
+        that can be returned.
+        """
+        self._labels = labels
+
+    @staticmethod
+    def create(labels: Dict[str, str]) -> "Resource":
+        """create a new resource.
+
+        Args:
+            labels: the labels that define the resource
+
+        Returns:
+            The resource with the labels in question
+        """
+    @property
+    @abstractmethod
+    def labels(self) -> Dict[str, str]:
+        """Return the label dictionary associated with this resource.
+
+        Returns:
+            A dictionary with the labels of the resource
+        """
+    def merge(self, other: Union["Resource", None]) -> "Resource":
+        """Return a resource with the union of labels for both resources.
+
+        Labels that exist in the main Resource take
+        precedence unless the label value is the empty string.
+
+        Args:
+            other: the resource to merge in
```

Comment:
```suggestion
            other: The resource to merge in.
```

Or lowercase and lose the period in other docstrings. Same thing for the other args/returns in this PR.

Reply:
sorry I'll try to be consistent in the future. Looking at Sphinx, it lists an example of Google style:

http://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#google-vs-numpy

Which capitalizes without a period for arguments.